From daee5ea768f8fdbace48da97ca8be091d9e88f54 Mon Sep 17 00:00:00 2001
From: David S. Wang <dsw@cloudera.com>
Date: Wed, 27 Mar 2013 15:01:51 -0700
Subject: [PATCH 32/96] HBASE-7526 create table does not log the table name in audit log

Reason: Improvement
Author: Matteo Bertozzi
Ref: CDH-9633
(cherry picked from commit f8d7b6d6c2da3d4a5a425ddbcecf99ae4b3e1ed6)
---
 .../hbase/security/access/AccessController.java    |   17 +++++++----------
 1 files changed, 7 insertions(+), 10 deletions(-)

diff --git a/security/src/main/java/org/apache/hadoop/hbase/security/access/AccessController.java b/security/src/main/java/org/apache/hadoop/hbase/security/access/AccessController.java
index af6ef4c..1a13eba 100644
--- a/security/src/main/java/org/apache/hadoop/hbase/security/access/AccessController.java
+++ b/security/src/main/java/org/apache/hadoop/hbase/security/access/AccessController.java
@@ -356,15 +356,7 @@ public class AccessController extends BaseRegionObserver
    * @throws AccessDeniedException if authorization is denied
    */
   private void requirePermission(String request, Permission.Action perm) throws IOException {
-    User user = getActiveUser();
-    if (authManager.authorize(user, perm)) {
-      logResult(AuthResult.allow(request, "Global check allowed", user, perm, null, null));
-    } else {
-      logResult(AuthResult.deny(request, "Global check failed", user, perm, null, null));
-      throw new AccessDeniedException("Insufficient permissions for user '" +
-          (user != null ? user.getShortName() : "null") +"' (global, action=" +
-          perm.toString() + ")");
-    }
+    requireGlobalPermission(request, perm, null, null);
   }
 
   /**
@@ -491,7 +483,12 @@ public class AccessController extends BaseRegionObserver
   @Override
   public void preCreateTable(ObserverContext<MasterCoprocessorEnvironment> c,
       HTableDescriptor desc, HRegionInfo[] regions) throws IOException {
-    requirePermission("createTable", Permission.Action.CREATE);
+    Set<byte[]> families = desc.getFamiliesKeys();
+    HashMap<byte[], Set<byte[]>> familyMap = Maps.newHashMapWithExpectedSize(families.size());
+    for (byte[] family: families) {
+      familyMap.put(family, null);
+    }
+    requireGlobalPermission("createTable", Permission.Action.CREATE, desc.getName(), familyMap);
   }
 
   @Override
-- 
1.7.0.4

